frontend/src/app/search.tsx

```diff
@@ -178,41 +203,42 @@ export default class Search extends Component {
     );
   }
     
+  onChangePage(newPage){
+    this.setState({page: newPage},()=>{console.log("New page number: "+this.state.page)})
+  }
 
   private handleSelectAll = (event) => {
     console.log('handleSelectAll')
     this.setState({check: !this.state.check}, () => {
       this.setState(prevState => {
+        this.state.allPaths=[]
+        this.transientPaths=[]
         const selectAllcheck = this.state.data.map(dataitem => {
               dataitem["checkedItem"] = this.state.check
               console.log(dataitem["pant:transientPath"]+":"+dataitem["checkedItem"])
-              this.state.check?this.state.allPaths.push(dataitem["pant:transientPath"]):delete this.state.allPaths[this.state.allPaths.indexOf(dataitem["pant:transientPath"])]
+              this.state.check?this.state.allPaths.push(dataitem["pant:transientPath"]):this.state.allPaths.splice(this.state.allPaths.indexOf(dataitem["pant:transientPath"]),1)
```

Comment:
I see some of the checkbox changes here, make sure you don't overwrite your checkboxes changes that are already on Master. When doing the rebase the new change set should only show the stuff for the pagination.